internal/strace/strace.go

```diff
@@ -45,25 +46,27 @@ var (
 
 	// This regex parses just the file path. Bytes written is parsed further below as the nature of the write buffer makes it unideal to parse through regex.
 	// TODO: We can see how we can potentially reuse regex patterns.
+	// I0928 00:18:54.794008     365 strace.go:593] [   6:   6] uname E write(0x1 pipe:[5], 0x555695ceaab0 "Linux 4.4.0\n", 0xc)
 	writePattern = regexp.MustCompile(`\S+ ([^,]+),.*`)
 )
 
 // We expect bytes written in the write syscall to be in hex.
 const hexPrefix = "0x"
 
 type FileInfo struct {
-	Path      string
-	Read      bool
-	Write     bool
-	Delete    bool
-	WriteInfo WriteInfo
+	Path             string
+	Read             bool
+	Write            bool
+	Delete           bool
+	WriteInfo        WriteInfo
+	WriteBufferPaths []string
 }
 
 type WriteInfo []WriteContentInfo
 
 type WriteContentInfo struct {
-	// TODO: A future PR will add to the WriteContentInfo struct a reference to a file. That file referenced will save the contents of the write buffer.
-	BytesWritten int64
+	WriteBufferId string
```

Comment:
Can we just move the buffer paths here? 

It consolidates the data a bit better and this way we also avoid another sha256sum when it comes time to upload. 

We can avoid this getting serialized to JSON via a custom tag: https://pkg.go.dev/encoding/json#Marshal:

```go
// Field is ignored by this package.
Field int `json:"-"`
```

Reply:
Is this what you mean? 

WriteContentInfo struct {
WriteBufferId string
WriteBufferPath string
}

(Note that the above two fields are different because of the random value added as a result of os.CreateTemp. I do have some questions about permissions below and if that is resolved we could potentially use os.WriteFile in which cases these can be the same value)

And in that case would we remove FileWriteBufferPaths from the DynamicAnalaysisResults struct? Because that would require us to dig into the strace data structures to get the write buffer path. 

I'm also not sure what you mean by avoid another sha256sum when it comes time to upload. Do you mean grab the WriteBufferId from the WriteContentInfo struct instead of computing the sha256 hash from the write buffer we retrieve? Is computing a sha256 hash expensive? I was trying to avoid breaking abstraction and accessing the strace structs.

Reply:
What you have here might actually be better if we want to go with batch uploading of file buffers. Feel free to ignore this comment for now.